qcfractal/interface/collections/openffworkflow.py

```diff
@@ -155,6 +151,33 @@ def add_fragment(self, fragment_id, data, provenance={}):
         # Push collection data back to server
         self.save(overwrite=True)
 
+    def _add_torsiondrive(self, packet):
+        # Build out a new service
+        torsion_meta = copy.deepcopy(
+            {k: self.data["torsiondrive_static_options"][k]
+                for k in ("torsiondrive_meta", "optimization_meta",  "qc_meta")})
+
+        for k in ["grid_spacing", "dihedrals"]:
+            torsion_meta["torsiondrive_meta"][k] = packet[k]
+
+        # Get hash of torsion
+        ret = self.client.add_service("torsiondrive", [packet["initial_molecule"]], torsion_meta)
+        return ret
+
+    def _add_optimize(self, packet):
+
+        optimization_meta = copy.deepcopy(
+            {k: self.data["optimization_static_options"][k]
+             for k in ("optimization_meta", "qc_meta")})
+
+        for k in ["constraints"]:
+            optimization_meta["optimization_meta"][k] = packet[k]
+
+        # Get hash of optimization
+        ret = self.client.add_procedure("optimization", "geoemetric", optimization_meta, [packet["initial_molecule"]])
```

Comment:
Did you want to allow flexibility for folks to choose something besides geometric here? This would need to go into the metadata.